### ISSUE TITLE
Log the epoch

### ISSUE THREAD

**Issue by robmarkcole** [ultralytics/ultralytics#20717]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

When training in a notebook, after around 120 epochs the logs are no longer visible, and it is necessary to rely on a remote logger (e.g. mlflow) to monitor progress. However the epoch is not logged, so overall progress is hard to determine. This FR is to log the epoch, so progress can be easily monitored

<img width="572" alt="Image" src="https://github.com/user-attachments/assets/9cc3307f-5cfd-469f-9c1c-080a18f5b30c" />

Would like to see epoch below:

<img width="475" alt="Image" src="https://github.com/user-attachments/assets/75cfff84-3d99-415b-b57b-1cffebb678f5" />

### Additional

The step count could alternatively be monitored:

<img width="615" alt="Image" src="https://github.com/user-attachments/assets/f5db786a-0182-48bd-9688-5b7b49a75d2a" />

Which derives from the epoch: `mlflow.log_metrics(..., step=trainer.epoch)`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @robmarkcole, thank you for your suggestion and for helping improve Ultralytics 🚀! Your feature request to log the epoch for better progress monitoring, especially when using remote loggers in notebooks, is much appreciated.

This is an automated response to help you get started. An Ultralytics engineer will review your request and assist you further soon.

If you have additional context, code snippets, or would like to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to illustrate how you’d like the epoch logged, please add it here. This will help us better understand your workflow and ensure the feature meets your needs.

In the meantime, you can explore our [Docs](https://docs.ultralytics.com/), which include many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. For custom training questions or more details about your setup, sharing dataset samples and training logs is always helpful.

Join the Ultralytics community for more discussion! For real-time chat, check out [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth topics, visit [Discourse](https://community.ultralytics.com/), or share your ideas on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Make sure you're using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your suggestion!

**Comment by robmarkcole** [comment]
@Y-T-G yes it is, although curiously it differs for wandb `wb.run.log(trainer.lr, step=trainer.epoch + 1)`

So knowing it can be inferred from the step count, the benefits of logging directly would be:
- See on the main metric table
- Clarity of the precise epoch, not depending on the logger chosen
